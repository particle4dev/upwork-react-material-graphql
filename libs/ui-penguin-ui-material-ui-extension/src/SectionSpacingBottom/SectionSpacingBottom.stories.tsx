import * as React from 'react';
import Typography from '@material-ui/core/Typography';
import { withKnobs } from "@storybook/addon-knobs";

import SectionSpacingBottom from './index';

export default {
  title: "SectionSpacingBottom",
  decorators: [withKnobs]
};

export const Basic = () => <div style={{backgroundColor: 'red'}}>
  <Typography>Basic 1</Typography>
  <SectionSpacingBottom/>
  <Typography>Basic 2</Typography>
</div>;
