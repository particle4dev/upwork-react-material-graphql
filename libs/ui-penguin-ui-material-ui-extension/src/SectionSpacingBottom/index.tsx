import * as React from 'react';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const debug = require('debug')('@penguin-ui/material-ui-extension:SectionSpacingBottom');

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      marginBottom: theme.spacing(2),
      height: 0
    }
  }),
);

export type SectionSpacingBottomProps = {
  style?: React.CSSProperties
}

const SectionSpacingBottom = React.forwardRef(function SectionSpacingBottom(props: SectionSpacingBottomProps, ref: React.Ref<HTMLDivElement>) {
  debug('render');

  const classes = useStyles();

  return (
    <div className={classes.root} ref={ref} {...props}>&nbsp;</div>
  );
});

if(process.env.NODE_ENV !== 'production') {
  SectionSpacingBottom.displayName = 'penguin_ui_material_ui_extension__SectionSpacingBottom';
}

SectionSpacingBottom.defaultProps = {};

export default SectionSpacingBottom;
