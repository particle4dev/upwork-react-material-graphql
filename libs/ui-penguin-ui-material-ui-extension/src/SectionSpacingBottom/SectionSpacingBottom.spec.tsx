import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import SectionSpacingBottom from './index';

describe('SectionSpacingBottom', () => {
  it('should render successfully', () => {
    const { baseElement, getByText } = render(<SectionSpacingBottom />);
    expect(baseElement).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
