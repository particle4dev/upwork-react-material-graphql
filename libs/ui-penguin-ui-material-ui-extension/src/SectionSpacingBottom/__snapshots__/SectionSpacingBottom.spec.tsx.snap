// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionSpacingBottom should render successfully 1`] = `
<body>
  <div>
    <div
      class="makeStyles-root-1"
    >
      Â 
    </div>
  </div>
</body>
`;
